app/src/main/java/com/example/triptracker/view/home/HomeScreen.kt

```diff
@@ -302,6 +319,132 @@ fun SearchBarImplementation(
   }
 }
 
+@Composable
+fun DisplayItineraryFromCategory(
+    itineraries: List<Itinerary>,
+    navigation: Navigation,
+    homeViewModel: HomeViewModel,
+    category: HomeCategory,
+    test: Boolean = false
+) {
+  LazyColumn(
+      modifier =
+          Modifier.fillMaxSize()
+              .padding(
+                  PaddingValues(
+                      0.dp, 0.dp, 0.dp, 70.dp)) // this ensures having a padding at the bottom
+              .testTag("ItineraryList"),
+      contentPadding = PaddingValues(16.dp)) {
+        items(itineraries) { itinerary ->
+          Log.d("ItineraryToDisplay", "Displaying itinerary: $itinerary")
+          DisplayItinerary(
+              itinerary = itinerary,
+              onClick = {
+                navigation.navigateTo(Route.MAPS, itinerary.id)
+                homeViewModel.incrementClickCount(itinerary.id)
+              },
+              displayImage = true,
+              test = test,
+          )
+        }
+      }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun HomePager(
+    itineraries: List<Itinerary>,
+    navigation: Navigation,
+    profile: MutableUserProfile,
+    homeViewModel: HomeViewModel = viewModel(),
+    innerPadding: PaddingValues,
+    test: Boolean = false
+) {
+  val ambientProfile = AmbientUserProfile.current
+  val userEmail = ambientProfile.userProfile.value.mail
+  val tabs = listOf(HomeCategory.TRENDING.name, HomeCategory.FOLLOWING.name)
+  val pagerState =
+      rememberPagerState(initialPage = 0) {
+        2 // initial page is 0, trending tab
+      }
+  var selectedTab by remember { mutableStateOf(pagerState.currentPage) }
+  var isSelected by remember { mutableStateOf(false) }
+
+  // added for page animation, if the animation is not smooth, change to pagerState.scrollToPage
+  LaunchedEffect(key1 = selectedTab) {
+    pagerState.animateScrollToPage(page = selectedTab)
+    when (selectedTab) {
+      0 -> homeViewModel.filterByTrending()
+      1 -> homeViewModel.filterByFollowing(userEmail)
+    }
+  }
+
+  // LaunchedEffect to synchronize the pager state with the selected tab
+  LaunchedEffect(pagerState.currentPage) {
+    selectedTab = pagerState.currentPage
+    when (pagerState.currentPage) {
+      0 -> homeViewModel.filterByTrending()
+      1 -> homeViewModel.filterByFollowing(userEmail)
+    }
+  }
+
+  Column(modifier = Modifier.background(md_theme_light_onPrimary)) {
+    TabRow(
+        selectedTabIndex = pagerState.currentPage,
+        modifier = Modifier.fillMaxWidth().height(50.dp),
+        backgroundColor = md_theme_light_onPrimary) {
+          tabs.forEachIndexed { index, title ->
+            isSelected = index == selectedTab
+            Tab(selected = isSelected, onClick = { selectedTab = index }) {
+              Text(
+                  title,
+                  color = if (isSelected) md_theme_light_black else md_theme_grey,
```

Comment:
This color will not work in dark mode I think. We should use the implementation Pol mentionned for the text in dark mode.